commons/zenoh-protocol/src/core/cowstr.rs

```diff
@@ -0,0 +1,105 @@
+use core::fmt::{Debug, Display, Formatter};
+use core::num::NonZeroUsize;
+
+enum CowStrInner<'a> {
+    Borrowed(&'a str),
+    Owned { s: Box<str>, capacity: NonZeroUsize },
```

Comment:
If I got correctly, an empty string can be represented by a `Borrowed("")`, right?

Reply:
Correct